independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/QuarkusBootstrap.java

```diff
@@ -486,6 +486,8 @@ public Builder addClassLoaderEventListeners(List<ClassLoaderEventListener> class
     public enum Mode {
         DEV,
         TEST,
-        PROD;
+        PROD,
+        REMOTE_DEV_SERVER,
+        REMOTE_DEV_CLIENT;
```

Comment:
I was wondering whether the LaunchMode could be enhanced in this manner as well. That could make it less confusing for remote dev.

Reply:
I thought about this, but there is already lots of code that tests for specific launch modes. If we add additional dev mode types any existing code that checks if launchmode == DEV will break in remote dev mode.

Reply:
Isn't the remote dev mode represented as NORMAL? isn't it supposed to be checked by calling `isRemoteDev()` currently? DEVELOPMENT would remain as is.

Reply:
REMOTE_DEV_CLIENT is normal, REMOTE_DEV_SERVER is DEVELOPMENT

Reply:
Would adding only the REMOTE_DEV_CLIENT to LaunchMode and keeping DEVELOPMENT for actual dev mode launch be a bit less confusing? I don't see `DevModeType.REMOTE_SERVER_SIDE` being checked anywhere.

Reply:
If we add an extra LaunchMode we need to update everywhere that LaunchMode is used in every extension, including downstream platform extensions. Because remote dev is not heavily tested in isolation I think the end result would be lots of extensions not handling the REMOTE_DEV_CLIENT mode properly and remote dev being very broken.

This is not really a different launch mode, the client generates a production app, the server runs a development app, the only place where this really matters is in how you handle what happens after the app is generated (i.e. deployment/container image building).